wren.go

```diff
@@ -318,6 +344,51 @@ func write(vm *C.WrenVM, text *C.char) {
 	fmt.Fprint(out, C.GoString(text))
 }
 
+//helper
+func readModule(dir string, name string) (string, error) {
+	var err error
+
+	// Precedence (dir/name.wren) next (dir/name/module.wren)
+	filename := filepath.Join(dir, name+".wren")
+	if _, err = os.Stat(filename); os.IsNotExist(err) {
+		filename = filepath.Join(dir, name, "module.wren")
+		if _, err = os.Stat(filename); os.IsNotExist(err) {
+			return "", err
+		}
+	}
+	var data []byte
+	if data, err = ioutil.ReadFile(filename); err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
```

Comment:
Thinking it through, I think it would make more sense for this method to optimistically read the file, and then

1. If no error, return the result
2. If an error other than "file not found", return the error
3. Otherwise, try the next file

File precedence can also be made a little more clear by defining it as a list:

```suggestion
func readModule(dir, name string) (string, error) {
	for _, filename := range []string{
		filepath.Join(dir, name+".wren"),
		filepath.Join(dir, name, "module.wren"),
	} {
		if data, err := ioutil.ReadFile(filename); err == nil {
			return string(data), nil
		} else if !os.IsNotExist(err) {
			return "", fmt.Errorf("load module: error reading file %s: %w", filename, err)
		}
	}
	return "", fmt.Errorf("module not found: %s", name) // could use a new error type, such as ErrModuleNotFound
}
```

Reply:
Refactored as per suggestion.